hivemq-edge/src/frontend/src/components/MQTT/EntityCreatableSelect.tsx

```diff
@@ -0,0 +1,221 @@
+
+export enum SelectEntityType {
+  TOPIC = 'TOPIC',
+  TOPIC_FILTER = 'TOPIC_FILTER',
+  TAG = 'TAG',
+}
+
+interface EntityOption extends OptionBase {
+  label: string
+  value: string
+  description?: string
+  // __isNew__ is not exposed in the OptionBase
+  __isNew__?: boolean
+}
+
+const customComponents = (
+  isMulti: boolean,
+  type: SelectEntityType
+): SelectComponentsConfig<EntityOption, boolean, GroupBase<EntityOption>> => ({
+  DropdownIndicator: null,
+  Option: ({ children, ...props }) => {
+    const { __isNew__ } = props.data
+    return (
+      <chakraComponents.Option {...props}>
+        <HStack>
+          {type === SelectEntityType.TOPIC && <TopicIcon mr={2} h={5} w={5} data-option-type={type} />}
+          {type === SelectEntityType.TAG && <PLCTagIcon mr={2} h={5} w={5} data-option-type={type} />}
+          {type === SelectEntityType.TOPIC_FILTER && <TopicFilterIcon mr={2} h={5} w={5} data-option-type={type} />}
+          <VStack gap={0} alignItems="flex-start">
+            {!__isNew__ && <code>{props.data.label}</code>}
+            {__isNew__ && <Text>{props.data.label}</Text>}
+            <Text fontSize="sm">{props.data.description}</Text>
+          </VStack>
+        </HStack>
+      </chakraComponents.Option>
+    )
+  },
+
+  Control: ({ children, ...props }) => (
+    <chakraComponents.Control {...props}>
+      {!isMulti && type === SelectEntityType.TOPIC && <TopicIcon mr={0} ml={3} data-option-type={type} />}
+      {!isMulti && type === SelectEntityType.TAG && <PLCTagIcon mr={0} ml={3} data-option-type={type} />}
+      {!isMulti && type === SelectEntityType.TOPIC_FILTER && <TopicFilterIcon mr={0} ml={3} data-option-type={type} />}
+      {children}
+    </chakraComponents.Control>
+  ),
+
+  MultiValueContainer: ({ children, ...props }) => {
+    return (
+      // TODO[NVL] removing the default style might cause side effects
+      <chakraComponents.MultiValueContainer {...props} sx={{}}>
+        {type === SelectEntityType.TOPIC && <Topic tagTitle={children} mr={3} />}
+        {type === SelectEntityType.TAG && <PLCTag tagTitle={children} mr={3} />}
+        {type === SelectEntityType.TOPIC_FILTER && <TopicFilter tagTitle={children} mr={3} />}
+      </chakraComponents.MultiValueContainer>
+    )
+  },
+})
+
+interface EntitySelectProps {
+  id?: string
+  isCreatable?: boolean
+  isMulti?: boolean
+  type: SelectEntityType
+  isLoading: boolean
+  options: EntityOption[]
+  icon: FC<IconProps>
+  onChange: (newValue: OnChangeValue<string, boolean>) => void
+  value: PropsValue<string>
+}
+
+const EntityCreatableSelect: FC<EntitySelectProps> = ({
+  type,
+  isCreatable = false,
+  isMulti = false,
+  options,
+  isLoading,
+  id,
+  onChange,
+  value,
+}) => {
+  const { t } = useTranslation('components')
+  const safeValue = useMemo(() => {
+    if (value === null) return null
+    if (isMulti) return (value as MultiValue<string>).map((w: string) => ({ label: w, value: w }))
```

Comment:
suggestion: `w` is unclear as a variable to me

```suggestion
    if (isMulti) return (value as MultiValue<string>).map((singleValue: string) => ({ label: singleValue, value: singleValue }))
```